qiskit_experiments/analysis/curve_analysis.py

```diff
@@ -0,0 +1,681 @@
+        however the parameters are shared with both functions.
+
+        .. code-block::
+
+            class AnalysisExample(CurveAnalysis):
+
+                __x_key__ = "delays"
+
+                __series__ = [
+                    SeriesDef(
+                        name="x",
+                        param_names=["a", "freq", "phase", "b"],
+                        fit_func_index=0,
+                        filter_kwargs={"pulse": "x"}
+                    ),
+                    SeriesDef(
+                        name="y",
+                        param_names=["a", "freq", "phase", "b"],
+                        fit_func_index=1,
+                        filter_kwargs={"pulse": "y"}
+                    )
+                ]
+
+                __fit_funcs__ = [library.cos, library.sin]
+
+                __param_names__ = ["a", "freq", "phase", "b"]
+
+    .. notes::
+        This class provides several private methods that subclasses can override.
+
+        - _run_fitting: Central method to perform fitting with the provided list of curve data.
+            Subclasses can create initial guesses of parameters and override
+            fitter analysis options here.
+            Note that each curve data provides circuit metadata that may be useful to
+            calculate initial guesses or apply some coefficients to values.
+
+        - _create_figure: A method to create figures. Subclasses can override this method
+            to create figures. Both raw data and fit analysis are provided.
+
+        - _data_processing: A method to format a list of circuit results into y and yerr values.
+            Subclasses can override this method depending on the expected circuit result format
+            and the output data representation.
+
+        - _post_processing: A method to calculate a new entity from a fit result.
+            This returns the fit result as-is by default.
+    """
+
+    #: str: Metadata key representing a scanned value.
+    __x_key__ = "xval"
+
+    #: List[SeriesDef]: List of mapping representing a data series
+    __series__ = None
+
+    #: List[Callable]: A callback function to define the expected curve
+    __fit_funcs__ = None
+
+    #: List[str]: Parameter name list
+    __param_names__ = list()
+
+    # Callable: Default curve fitter. This can be overwritten.
+    __base_fitter__ = scipy_curve_fit_wrapper
+
+    def _run_fitting(self, curve_data: List[CurveEntry], **options) -> AnalysisResult:
+        """Fit series of curve data.
+
+        Subclass can override this method to provide initial guess or fit boundaries.
+        For example, initial guess is not automatically provided by this base class.
+
+        Args:
+            curve_data: List of raw curve data points to fit.
+            **options: Fitting options.
+
+        Returns:
+            Analysis result populated by fit parameters.
+        """
+        return self._series_curve_fit(curve_data=curve_data, **options)
```

Comment:
You're right. I merged `_series_curve_fit` into `_run_fitting`, and added `_setup_fitting` and `_data_pre_processing` method. @gadial 's RB analysis code requires these logics. Actually I intended to do these pre-processing in `_run_fitting`, but this might be bit confusing.